crates/typst/src/layout/flow.rs

```diff
@@ -856,6 +960,44 @@ impl<'a, 'e> FlowLayouter<'a, 'e> {
 
         Ok(())
     }
+
+    fn layout_line_number(&mut self, numbering: &Numbering, y: Abs) -> SourceResult<()> {
+        let line_counter = Counter::of(ParLineMarker::elem());
+        let mut line_counter_update = line_counter
+            .clone()
+            .update(Span::detached(), CounterUpdate::Step(NonZeroUsize::ONE));
+
+        let update_hash = crate::utils::hash128(&line_counter_update);
+        let location = self.locator.next_location(self.engine.introspector, update_hash);
+        line_counter_update.set_location(location);
```

Comment:
This manual location management is a bit unidiomatic. `next_location` is not supposed to be used outside of the core realization code. 

I don't think it's necessary. Wouldn't just a normal `update() + display()` work?

Reply:
You can replace `display_at_loc` with `CounterDisplayElem` to not require a location.